src/Compilers/Core/Portable/PortableShim.cs

```diff
@@ -495,11 +495,19 @@ internal static class Encoding
 
         internal static class MemoryStream
         {
+            internal delegate bool TryGetBufferDelegate(System.IO.MemoryStream stream, out ArraySegment<byte> buffer);
+
             internal static readonly Type Type = typeof(System.IO.MemoryStream);
 
-            internal static readonly MethodInfo GetBuffer = Type
+            internal static readonly Func<System.IO.MemoryStream, byte[]> GetBuffer = Type
+                .GetTypeInfo()
+                .GetDeclaredMethod(nameof(GetBuffer))
+                .CreateDelegate<Func<System.IO.MemoryStream, byte[]>>();
+
+            internal static readonly TryGetBufferDelegate TryGetBuffer = Type
                 .GetTypeInfo()
-                .GetDeclaredMethod(nameof(GetBuffer));
+                .GetDeclaredMethod(nameof(TryGetBuffer))
+                .CreateDelegate<TryGetBufferDelegate>();
```

Comment:
@nguerrera if you're wondering what terrible person would implement an extension method that handles `null` values for `this` ... that was me.  I'm still a bit ashamed.